numpy/fft/_helper.py

```diff
@@ -163,10 +167,9 @@ def fftfreq(n, d=1.0, device=None):
     >>> freq = np.fft.fftfreq(n, d=timestep)
     >>> freq
     array([ 0.  ,  1.25,  2.5 , ..., -3.75, -2.5 , -1.25])
-
     """
-    if not isinstance(n, integer_types):
-        raise ValueError("n should be an integer")
+    if not isinstance(n, integer_types) or n <= 0:
```

Comment:
These days, it would actually be better to just use:
```
n = operator.index(n)
```
and then check `n <= 0`.  (And that changes the error type to `TypeError` for many things, which I think is completely fine or actually better).